server/server_test.go

```diff
@@ -4,82 +4,45 @@
 package server
 
 import (
-	"flag"
 	"io/ioutil"
 	"net/http"
 	"net/http/httptest"
-	"os"
 	"path/filepath"
-	"strings"
 	"testing"
+	"time"
 
+	"github.com/stretchr/testify/assert"
 	"github.com/stretchr/testify/require"
 
-	"github.com/mattermost/mattermost-server/v5/mlog"
 	"github.com/xeipuuv/gojsonschema"
 )
 
-var config *Config
-var err error
-var s *Server
-
-func TestMain(m *testing.M) {
-	var configFile string
-	flag.StringVar(&configFile, "config", "config-mattermod.default.json", "")
-	flag.Parse()
-	config, err = GetConfig(configFile)
-	if err != nil {
-		panic(err)
-	}
-	SetupLogging(config)
-	mlog.Info("Loaded config", mlog.String("filename", configFile))
-
-	s, err = New(config)
-	if err != nil {
-		panic(err)
-	}
-	s.Start()
-
-	exitVal := m.Run()
-
-	os.Exit(exitVal)
-}
-
 func TestPing(t *testing.T) {
-	defer s.Stop()
-
-	split := strings.Split(config.ListenAddress, ":")
-	require.Len(t, split, 2)
+	s := &Server{
+		StartTime: time.Now(),
+	}
 
-	url := "http://localhost:" + split[1]
+	ts := httptest.NewServer(http.HandlerFunc(s.ping))
+	defer ts.Close()
 
-	req, err := http.NewRequest(http.MethodGet, url, nil)
+	res, err := http.Get(ts.URL)
 	require.NoError(t, err)
-	w := httptest.NewRecorder()
-	s.ping(w, req)
+	defer res.Body.Close()
 
-	body := w.Result().Body
-	defer body.Close()
-	bytes, err := ioutil.ReadAll(body)
+	bytes, err := ioutil.ReadAll(res.Body)
 	require.NoError(t, err)
 	bytesLoader := gojsonschema.NewBytesLoader(bytes)
 
 	dir, err := filepath.Abs(filepath.Dir(""))
-	if err != nil {
-		mlog.Error("unable to find project path")
-	}
-	jsonLoader := gojsonschema.NewReferenceLoader("file://" + dir + "/schema/ping.schema.json")
+	assert.NoError(t, err)
 
+	jsonLoader := gojsonschema.NewReferenceLoader("file://" + dir + "/schema/ping.schema.json")
```

Comment:
I thought the ticket was also about removing this pingschema.json. Any reason this is here? We should do a json unmarshal and verify the contents.

Reply:
I thought the test is more about *at least* ensuring the response is in a certain schema. 🤷 Updated the PR anyway..